schemas/typescript/types.ts

```diff
@@ -6,99 +6,196 @@
+   */
   kind: "the-compact" | "rhinestone";
-  /** Lock-specific parameters */
+  /** 
+   * Lock-specific parameters
+   * @description Additional configuration parameters specific to the chosen lock type
+   * @example For "the-compact": { "nonce": "123", "expiry": 1234567890 }
+   * @example For "rhinestone": { "module": "0xabc...", "data": "0x123..." }
+   */
   params?: Record<string, unknown>;
 }
 
 /**
  * Origin submission preference
- * @description Explicit, forward-compatible preference for who submits the origin transaction and acceptable authorization schemes.
+ * @description Specifies who submits the origin transaction and which authorization schemes are acceptable.
+ *              This allows flexible gas payment and signature management across different protocols.
  */
 export interface OriginSubmission {
-  /** Who submits the origin transaction */
+  /** 
+   * Who submits the origin transaction
+   * @description Determines transaction submission responsibility
+   * - "user": User directly submits and pays gas
+   * - "protocol": Protocol/relayer submits on user's behalf (gasless for user)
+   */
   mode: "user" | "protocol";
-  /** Acceptable signing/authorization schemes for interoperability */
+  /** 
+   * Acceptable signing/authorization schemes for interoperability
+   * @description List of supported authorization methods for the transaction
+   * - "erc-4337": Account abstraction (smart contract wallets)
+   * - "permit2": Uniswap's Permit2 for token approvals
+   * - "erc20-permit": EIP-2612 permit for gasless approvals
+   * - "eip-3009": Transfer with authorization (e.g., USDC)
+   * @example ["permit2", "erc20-permit"] - Accept either Permit2 or ERC20 permit
+   */
   schemes?: Array<"erc-4337" | "permit2" | "erc20-permit" | "eip-3009">;
 }
 
 // ============ Quote Types ============
 
 /**
  * Available input from a user
- * @description Available input from a user
+ * @description Specifies assets that a user is willing to provide as input for a swap or transfer.
+ *              Represents the "from" side of the transaction.
+ * @example Exact-input quote (user has 4000 USDC):
+ * {
+ *   user: "0x00010000010114D8DA6BF26964AF9D7EED9E03E53415D37AA96045", // Ethereum mainnet
+ *   asset: "0x000100000101A0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48", // USDC on Ethereum
+ *   amount: "4000000000", // Exactly 4000 USDC
+ *   lock: { kind: "the-compact" }
+ * }
+ * @example Exact-output quote (amount undefined, will be quoted):
+ * {
+ *   user: "0x00010000010114D8DA6BF26964AF9D7EED9E03E53415D37AA96045",
+ *   asset: "0x000100000101A0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48", // USDC
+ *   amount: undefined, // Provider will quote how much USDC needed
+ *   lock: { kind: "the-compact" }
+ * }
  */
-export interface AvailableInput {
-  /** User address */
+export interface Input {
+  /** 
+   * User address 
+   * @description The address providing the input assets
+   */
   user: Address;
```

Comment:
Same question as before in example context, Address is in plain Ethereum format. Are we considering it is Ethereum mainnet or we need additional optional field chainId?

Address maybe can be in format string | {address: "", chainId: 1}. This way we do not need to extend root objects with optional chainId field. In case string is used it is EIP-7930.